### PR TITLE
Allow restricted access routing on highway=service roads

### PR DESCRIPTION
**Pull request by karenzshea**
# Issue

Closes https://github.com/Project-OSRM/osrm-backend/issues/4667

There is a [whitelist](https://github.com/Project-OSRM/osrm-backend/blob/a8f3474996af9390eb7166dddd5f4c62bb0ae4c1/profiles/car.lua#L156-L170) of highway types on which we allow restricted access routing, i.e. roads that you can route onto if the route starts/ends on the road, but not as a shortcut through. (https://github.com/Project-OSRM/osrm-backend/pull/3721). 

This PR makes a profile adjustment that adds `highway=service` roads onto the whitelist of roads we will end or start a route on when it's tagged with [a restricted access tag](https://github.com/Project-OSRM/osrm-backend/blob/a8f3474996af9390eb7166dddd5f4c62bb0ae4c1/profiles/car.lua#L82-L87).

Anecdotally speaking this will allow:
- a ferry route to be found between Isle of Man and Liverpool where the end of the ferry leads directly onto a dock/pier road of some kind tagged with both `highway=service` and `access=destination`. 
- routing people who really need help within parking lots that have been tagged with any combination of `highway=service` and `access=destination|customer|private`

## Tasklist
-  [x] Do some manual checks around airport service road tagging
 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [x] add regression / cucumber cases (see docs/testing.md)
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: unit_tests/library/route.cpp [hunk 1]
```diff
@@ -69,13 +69,13 @@ BOOST_AUTO_TEST_CASE(test_route_same_coordinates_fixture)
                                                    json::Object{{
                                                        {"location", location},
                                                        {"bearing_before", 0},
-                                                       {"bearing_after", 58},
+                                                       {"bearing_after", 238},
```
Comment (karenzshea) [inline unit_tests/library/route.cpp]: This seems like a strange change ... I wouldn't think that a change to access tags in the profiles would result in different bearings in this tests?

**Comment by karenzshea** [comment]
### First change: allowing restricted access routing onto `highway=service` roads
[Westhafenstraße](https://www.openstreetmap.org/way/178594661) is tagged `highway=service` and `access=destination`

image
We allow routing onto it now...

image
... but not through it even if it would be "faster"

### Second change: disallowing restricted access routing onto `highway=service` `access=private` roads
... doesn't work yet

File: unit_tests/library/route.cpp (hunk 1 above)
Comment (TheMarex) [reply unit_tests/library/route.cpp]: Hrm maybe it snaps to a different road now?

**Comment by karenzshea** [comment]
Update ...

We currently do not support routing onto ways with the tag combination `highway=service` `access=private` and want to preserve part of this behavior, while allowing restricted access routing on other combinations of `highway=service` tagged roads.

The behavior we want to preserve is disallowing routes starting from `highway=service` `access=private` roads. This will require extra logic in the way parsing function that adds a boolean like `is_startpoint` or `can_start_from` that will later be used in the edge based graph factory to mark `highway=service` `access=private` roads as invalid start points.

**Comment by karenzshea** [comment]
And disallowing routing onto access=private service roads!
image
image

File: features/car/destination.feature [hunk 2]
```diff
@@ -131,3 +131,56 @@ Feature: Car - Destination only, no passing through
...
+            | cx    | private  | service |
+            | xd    | private  | service |
+            | de    |          | primary |
+
+        When I route I should get
+            | from | to | route     |
+            | a    | x  | acb,xd,xd |
+            | a    | d  | acb,xd,xd |
+            | a    | e  | acb,xd,de |
+            | x    | e  | de,de     |
```
Comment (TheMarex) [inline features/car/destination.feature]: A comment column would help, especially a comment that the last case depends on `x` snapping to `de`

**Comment by TheMarex** [comment]
@karenzshea You seem to have some failing test cases that need to be investigated:

https://ci.appveyor.com/project/DennisOSRM/osrm-backend/build/1.0.9836#L16366

**Comment by TheMarex** [comment]
Travis is just failing because of Mac OS X builds, merging this anyway.

**Pull request merged**